### PR TITLE
fix: Ignore keywords for uppercase=True

### PR DESCRIPTION
**Pull request by wugeer**
fix: https://github.com/shssoichiro/sqlformat-rs/issues/33

### CODE REVIEW

File: src/formatter.rs [hunk 1]
```diff
@@ -182,7 +186,15 @@ impl<'a> Formatter<'a> {
...
         }
-        if self.options.uppercase {
+        if self.options.uppercase
+            && (self.options.ignore_case_convert.is_none()
+                || !self
+                    .options
+                    .ignore_case_convert
+                    .as_ref()
+                    .unwrap()
+                    .contains(&token.value))
```
Comment (shssoichiro) [inline src/formatter.rs:196]: It's probably more idiomatic to do this with a `map` (apply to 2 more instances below as well)

```suggestion
            && !self
                    .options
                    .ignore_case_convert
                    .as_ref()
                    .map(|values| values.contains(&token.value))
                    .unwrap_or(false)
```

**Review by shssoichiro — APPROVED** [review]
Thanks!

**Review by BilelGho — COMMENTED** [review]
Hi. First time reviewer but I believe this is a breaking change and required a new major version.

File: src/lib.rs [hunk 2]
```diff
@@ -15,16 +15,16 @@ mod tokenizer;
 
 /// Formats whitespace in a SQL string to make it easier to read.
 /// Optionally replaces parameter placeholders with `params`.
-pub fn format(query: &str, params: &QueryParams, options: FormatOptions) -> String {
+pub fn format(query: &str, params: &QueryParams, options: &FormatOptions) -> String {
```
[the remark this answers is not in the document]
Comment (shssoichiro) [reply src/lib.rs:18]: The fact that a field was added to `FormatOptions` and `Copy` was removed from it makes it difficult to revert the breaking change without reverting this entire PR. I think I will move forward with releasing the new version as semver breaking instead. Thanks.

**Pull request merged**